core/src/main/java/org/apache/iceberg/SystemConfigs.java

```diff
@@ -42,6 +42,17 @@ private SystemConfigs() {}
           Math.max(2, Runtime.getRuntime().availableProcessors()),
           Integer::parseUnsignedInt);
 
+  /**
+   * The size of the queue in ParallelIterable. This queue limits the memory usage of manifest
+   * reader.
+   */
+  public static final ConfigEntry<Integer> MANIFESTS_READER_QUEUE_SIZE =
```

Comment:
The ParallelIterable is a common util, does this config is used target for manifest only?

Reply:
I only found used for read manifest.

Reply:
It doesn't matter that it is only used for one purpose right now. The iterable should be kept generic by passing in configuration.